HIP/hip-655.md

```diff
@@ -0,0 +1,106 @@
+- Airdrop-Recipient: the intended recipient of the above airdrop
+- Node-Operator: the operator of a node in the Hedera network
+    
+### User stories
+    
+- As an Airdrop-Sender, I want to send arbitrary tokens (Fungible and/or Non-fungible) to Airdrop-Recipients so that I can create interesting business use cases
+    1. I understand that the recipients of the airdrop may not wish to accept these tokens and I am willing to pay for the network resources used while they make their decision.
+    2. If the intended Airdrop-Recipients do not accept my transfer, I want to be able to restore those tokens to my account so that I can do whatever I wish with those tokens.
+- As an Airdrop-Recipient, I want to receive arbitrary tokens (Fungible and/or Non-fungible) from the Airdrop-Senders so that I can accumulate those in my account.
+    1. I want to achieve the above while having explicit control over what tokens actually show up in my account.
+    2. I want to achieve the above without being subject to the reputational/legal/spam risk of being associated with unwanted tokens
+    3. I want to achieve the above without having to be forced to pay for the storage/renewal of these unwanted tokens unless I explicitly accept those.
+- As a node operator in the network, I want to make sure that the network security is not compromised and that somebody pays for the network resources that are used. The design of this HIP is partially inspired by how we handle spam emails in today's world, but with an added safeguard of ensuring that there is an economic disincentive for an Airdrop-Sender from sending indiscriminate spam. Through this HIP, we will create a new concept of unsolicited-token-outbox/inbox that maintain a record of intended transfers of airdrops.
+    
+## Specification
+    
+- Airdrop-Sender sends (i.e. airdrops) tokens to a list of Airdrop-Recipients using a regular `CryptoTransfer` call. For this discussion, let us assume that none of the intended Airdrop-Recipients have associated this token with their accounts.
+- During token transfer validation, the network finds that the receivers have not associated this token with their accounts. Instead of rejecting this call with a failure code, the new behavior will store these tokens in the Airdrop-Sender's account in a new data structure called the `unsolicited-tokens-outbox`. Similarly, there is a concept of an `unsolicited-tokens-inbox` (or the incoming spam inbox) in the Airdrop-Recipients' accounts. And an entry is added in each of these recipient accounts registering the proposed transfers.
+- Network resources are used for storing these entries in both of these accounts (sender and receiver). The `CryptoTransfer` call will have an additional fee to cover the rent for these resources for one auto-renewal cycle. And the Airdrop-Sender will have to pay for the renewal of these resources in their normal auto-renewal process while these tokens remain in the unsolicited-outbox.
```

Comment:
Is there a size limit to this list in terms of how many tokens can be in a potential state of acceptance for a user?